src/main/java/school/lemon/changerequest/java/extendedinteger/ExtendedInteger.java

```diff
@@ -4,116 +4,116 @@
  * Integer representation with some additional features
  */
 public class ExtendedInteger {
+    private int value;
 
     public ExtendedInteger(int value) {
-        //TODO: implement me
+        this.value = value;
     }
 
-    /**
-     * Check whether specified value is even
-     * @param value to check
-     * @return true if value is even, false - otherwise
-     */
+
     public static boolean isEven(int value) {
-        //TODO: implement me
-        return false;
+        return ((value & 1) == 0);
     }
 
-    /**
-     * Check whether specified value is odd
-     * @param value to check
-     * @return true if value is odd, false - otherwise
-     */
     public static boolean isOdd(int value) {
-        //TODO: implement me
-        return false;
+        return (value % 2 != 0);
     }
 
-    /**
-     * Check whether specified value is prime
-     * @param value to check
-     * @return true if value is prime, false - otherwise
-     */
     public static boolean isPrime(int value) {
-        //TODO: implement me
-        return false;
+        if (value == 2) return true;
+        if (value < 2 || value % 2 == 0) return false;
+        for (int i = 2; i < value; i++) {
+            if (value % i == 0) {
+                return false;
+            }
+        }
+        return true;
     }
 
-    /**
-     * Parse specified char array and create instance of {@code ExtendedInteger}
-     * @param value to parse
-     * @return instance of {@code ExtendedInteger} or
-     * null in case specified value is null or the value does not contain a parsable integer
-     */
+
     public static ExtendedInteger parseInt(char[] value) {
-        //TODO: implement me
-        return null;
+        if (value == null || value.length == 0) return null;
+        String newValue = new String(value);
+        StringBuilder str = new StringBuilder();
```

Comment:
If you pass your char array to `String` - is there any purpose of re-implementing algorithm ? Cause you already have implementation for `String` argument below.
You can just call it and return it's result, like: `return parseInt(new String(value));`